### PR TITLE
PYIC-3581: Include mitigation details in audit event

### PR DESCRIPTION
**Pull request by Wynndow**
### What changed

Include mitigation details in audit event

### Why did it change

This adds a couple of bits of info to the `IPV_IDENTITY_ISSUED` audit event issued in the build-user-identity handler. This idea is to allow us to track how successful the mitigations work has been. For all identities issued we'll be able to see if the user has a mitigation. We'll also be able to see if they were breaching the CI threshold.

### Issue tracking

- [PYIC-3581](https://govukverify.atlassian.net/browse/PYIC-3581)

[PYIC-3581]: https://govukverify.atlassian.net/browse/PYIC-3581?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

#### Straight through success
```
{
    "event_name": "IPV_IDENTITY_ISSUED",
    "component_id": "https://dev-chrisw.01.dev.identity.account.gov.uk",
    "user": {
        "user_id": "urn:uuid:8123620e-a8b2-4b0d-950c-dbbb0b5ae263",
        "session_id": "AIrR-pl6jKRdi_vkgEd9loTDcd7UcDglPvMtaLJBIeE",
        "govuk_signin_journey_id": "47ca8cd7-aec2-4a77-b9ef-1973ef7ceca0",
        "ip_address": null
    },
    "extensions": {
        "levelOfConfidence": "P2",
        "ciFail": false,
        "hasMitigations": false
    },
    "timestamp": 1695132966
}
```
#### Existing identity (as above)
```
{
    "event_name": "IPV_IDENTITY_ISSUED",
    "component_id": "https://dev-chrisw.01.dev.identity.account.gov.uk",
    "user": {
        "user_id": "urn:uuid:8123620e-a8b2-4b0d-950c-dbbb0b5ae263",
        "session_id": "bI5Zn9RQatK2vkJqRQ-5UzJWxLbf2OD1DB_S07ERfPM",
        "govuk_signin_journey_id": "2140f35c-cd1a-44ea-ba8b-4bd879101f95",
        "ip_address": null
    },
    "extensions": {
        "levelOfConfidence": "P2",
        "ciFail": false,
        "hasMitigations": false
    },
    "timestamp": 1695133723
}
```
#### Failure with a CI
```
{
    "event_name": "IPV_IDENTITY_ISSUED",
    "component_id": "https://dev-chrisw.01.dev.identity.account.gov.uk",
    "user": {
        "user_id": "f2829474-6580-44bc-ae2c-042a1927f540",
        "session_id": "RJVZD8xFmmX_W95321u4lO67dXP6B7SsS2o1GNkNueQ",
        "govuk_signin_journey_id": "de17fb58-0909-4fd2-87ff-f5d3602c9443",
        "ip_address": null
    },
    "extensions": {
        "levelOfConfidence": "P0",
        "ciFail": true,
        "hasMitigations": false
    },
    "timestamp": 1695133591
}
```
#### Mitigating the previous failure
```
{
    "event_name": "IPV_IDENTITY_ISSUED",
    "component_id": "https://dev-chrisw.01.dev.identity.account.gov.uk",
    "user": {
        "user_id": "f2829474-6580-44bc-ae2c-042a1927f540",
        "session_id": "HpqSjeZp44O8sJJQHWj0VI0DfF5m_ILe-wPD4tXonRI",
        "govuk_signin_journey_id": "3d79e031-2f5e-4c0a-a7b3-035846a01458",
        "ip_address": null
    },
    "extensions": {
        "levelOfConfidence": "P2",
        "ciFail": false,
        "hasMitigations": true
    },
    "timestamp": 1695133659
}
```

### CODE REVIEW

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluator.java [hunk 1]
```diff
@@ -76,8 +80,12 @@ public Optional<JourneyResponse> getJourneyResponseForStoredContraIndicators(
                         .with(LOG_CI_SCORE.getFieldName(), ciScore));
 
         if (ciScore <= Integer.parseInt(configService.getSsmParameter(CI_SCORING_THRESHOLD))) {
+            ipvSession.setCiFail(false);
```
Comment (TalatAydinGDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluator.java:83]: I think, we can also add logs to the places where we set the 'ciFail' field to true, and we can assign an appropriate error code to it. Tracking this through the application logs in Splunk or CloudWatch would be easy.

**Review by TalatAydinGDS — COMMENTED** [review]
Looks good to me. I just added a recommend comment.

**Review by TalatAydinGDS — DISMISSED** [review]
Looks good to me. I just added a recommend comment.

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluator.java (hunk 1 above)
Comment (Wynndow) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/gpg45/Gpg45ProfileEvaluator.java:83]: We already have quite a lot of logging in the ci-scoring lambda around what scores the user has achieved, as well as the logs from the process-journey-event lambda which let us know what's happened to a user, so I'm inclined to not add more logging around this here.

**Review by TalatAydinGDS — APPROVED** [review]

**Pull request merged**